src/app/actions/posting.js

```
@@ -0,0 +1,32 @@
+import { endpoints } from '@r/api-client';
+
+const { PostsEndpoint } = endpoints;
+
+export const SUBREDDIT_SELECT = 'POSTING__SUBREDDIT_SELECT';
+export const FIELD_UPDATE = 'POSTING__FIELD_UPDATE';
+export const SUBMIT_PENDING = 'POSTING__SUBMIT_PENDING';
+export const SUBMIT_SUCCESS = 'POSTING__SUBMIT_SUCCESS';
+export const SUBMIT_FAILURE = 'POSTING__SUBMIT_FAILURE';
+
+export const selectSubreddit = subreddit => ({ type: SUBREDDIT_SELECT, subreddit });
+export const updateField = (field, value) => ({ type: FIELD_UPDATE, field, value });
+
+export const submitPost = data => async (dispatch) => {
+  // TODO: hardcode this so I don't accidentally post anywhere else
+  data.sr = 'PhilTestSubreddit';
+  data.sendreplies = true;
+  data.resubmit = true;     // how to handle these?
+
+  // TODO: captcha?
+  // captcha - user's response to captcha challenge
+  // iden    - id of captcha
+
+  dispatch({ type: SUBMIT_PENDING });
+  const response = await PostsEndpoint.post(data);
+
+  if (response.errors) {
```

So, something i want to try in modmail when i get a chance is to have the "api" throw specific errors for api stuff. that way, the catch block would look something like:

```es6
catch (e) {
  if (e instanceof ValidationError) {
    dispatch(formErrors(e));
  } else if (e instanceof ServerError) {
    dispatch(genericError(e));
  } else {
    throw e;
  }
}
```

That way, we can still separate error logic from the success error, but non-api related errors won't get swallowed